R/generate_fossils.R

```diff
@@ -39,6 +62,31 @@ png("assets/img/vertical_tree@1x.png", width = width, height = height)
 plot(tree, show.tip.label = FALSE, no.margin = TRUE)
 dev.off()
 
+output_cols <- list()
+for(i in 1:nrow(res)) {
+    row <- res[i,]
+    left <- str_replace(row$left, "_", " ")
+    right <- str_replace(row$right, "_", " ")
+    for (rank in wanted_ranks) {
+        splat <- split(tax, tax[[rank]])
+        for (named_rank in names(splat)) {
+            species <- output[[rank]][[named_rank]]$species
+            found <- FALSE
+            if(left %in% species && right %in% species) {
+                output_cols[[rank]] = c(output_cols[[rank]], named_rank)
+                found <- TRUE
+                break
+            }
+        }
+        if(!found) {
+            output_cols[[rank]] = c(output_cols[[rank]], NA)
+        }
+    }
+}
+
+for(rank in wanted_ranks) {
+    res <- cbind(res, output_cols[[rank]])
+}
```

Comment:
This creates multiple columns with the name `output_cols[[rank]]`. We can avoid this by just converting `output_cols` (which is a list) to a data frame.